src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetLibraryConverter.cs

```diff
@@ -0,0 +1,159 @@
+        private static readonly byte[] CompilePropertyName = Encoding.UTF8.GetBytes("compile");
+        private static readonly byte[] ResourcePropertyName = Encoding.UTF8.GetBytes("resource");
+        private static readonly byte[] NativePropertyName = Encoding.UTF8.GetBytes("native");
+        private static readonly byte[] BuildPropertyName = Encoding.UTF8.GetBytes("build");
+        private static readonly byte[] BuildMultiTargetingPropertyName = Encoding.UTF8.GetBytes("buildMultiTargeting");
+        private static readonly byte[] ContentFilesPropertyName = Encoding.UTF8.GetBytes("contentFiles");
+        private static readonly byte[] RuntimeTargetsPropertyName = Encoding.UTF8.GetBytes("runtimeTargets");
+        private static readonly byte[] ToolsPropertyName = Encoding.UTF8.GetBytes("tools");
+        private static readonly byte[] EmbedPropertyName = Encoding.UTF8.GetBytes("embed");
+        private static readonly byte[] FrameworkReferencesPropertyName = Encoding.UTF8.GetBytes("frameworkReferences");
+
+        public LockFileTargetLibrary Read(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException("Expected PropertyName, found " + reader.TokenType);
+            }
+
+            var lockFileTargetLibrary = new LockFileTargetLibrary();
+
+            //We want to read the property name right away
+            var propertyName = reader.GetString();
+            var (targetLibraryName, version) = propertyName.SplitInTwo('/');
+            lockFileTargetLibrary.Name = targetLibraryName;
+            lockFileTargetLibrary.Version = version is null ? null : JsonUtility.ParseNugetVersion(version);
+
+            reader.Read();
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException("Expected StartObject, found " + reader.TokenType);
+            }
+
+            while (reader.Read() && reader.TokenType == JsonTokenType.PropertyName)
+            {
+                if (reader.ValueTextEquals(TypePropertyName))
+                {
+                    lockFileTargetLibrary.Type = reader.ReadNextTokenAsString();
+                }
+                else if (reader.ValueTextEquals(FrameworkPropertyName))
+                {
+                    lockFileTargetLibrary.Framework = reader.ReadNextTokenAsString();
+                }
+                else if (reader.ValueTextEquals(DependenciesPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.Dependencies = ReadPackageDependencyList(ref reader);
+                }
+                else if (reader.ValueTextEquals(FrameworkAssembliesPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.FrameworkAssemblies = reader.ReadStringArrayAsIList(new List<string>());
```

Comment:
Is 'FrameworkAssemblies' ever empty? If so, figure out way to `Array.Empty<string>` instead of `new List<string>()` as the default.